contracts/messaging/dfk/bridge/HeroBridge.sol

```diff
@@ -0,0 +1,199 @@
+     * If any other value is returned or the interface is not implemented by the recipient, the transfer will be reverted.
+     *
+     * The selector can be obtained in Solidity with `onERC721Received.selector`.
+     */
+    function onERC721Received(
+        address operator,
+        address from,
+        uint256 tokenId,
+        bytes calldata data
+    ) external returns (bytes4) { 
+        return this.onERC721Received.selector;
+    }
+
+    function _createMessage(
+        uint256 _heroId,
+        address _dstUserAddress,
+        Hero memory _heroToBridge
+    ) internal pure returns (bytes memory) {
+        // create the message here from the nested struct
+        MessageFormat memory msgFormat = MessageFormat({
+            dstHeroId: _heroId,
+            dstHero: _heroToBridge,
+            dstUser: _dstUserAddress
+        });
+        return abi.encode(msgFormat);
+    }
+
+    function _decodeMessage(bytes memory _message)
+        internal
+        pure
+        returns (MessageFormat memory)
+    {
+        MessageFormat memory decodedMessage = abi.decode(
+            _message,
+            (MessageFormat)
+        );
+        return decodedMessage;
+    }
+
+    function _createOptions() public returns (bytes memory) {
+        abi.encodePacked(uint16(1), msgGasLimit);
+    }
+
+    /**
+     * @notice User must have an existing hero minted to bridge it.
+     * @param _heroId specifics which hero msg.sender already holds and will transfer to the bridge contract
+     * @param _dstChainId The destination chain ID - typically, standard EVM chain ID, but differs on nonEVM chains
+     */
+    function sendHero(uint256 _heroId, uint256 _dstChainId) external payable {
+        Hero memory heroToBridge = IHeroCoreUpgradeable(heroes).getHero(
+            _heroId
+        );
+        bytes32 receiver = trustedRemoteLookup[_dstChainId];
+        // _createMessage(heroId, dstUserAddress, Hero);
+        bytes memory msgToPass = _createMessage(
+            _heroId,
+            msg.sender,
+            heroToBridge
+        );
+        // Create _options
+        // temporarily empty
+        bytes memory options = _createOptions();
+
+        // revert if the hero is on a quest
+        require(
+            heroToBridge.state.currentQuest == address(0),
+            "hero is questing"
+        );
+
+        // revert if the hero is on auction
+        require(
+            !IAssistingAuction(assistingAuction).isOnAuction(_heroId),
+            "assisting auction"
+        );
+
+        IHeroCoreUpgradeable(heroes).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _heroId
+        );
+        // Hero now locked, message can be safely emitted
+
+        _send(receiver, _dstChainId, msgToPass, options);
+    }
+
+    // Function called by executeMessage() - handleMessage will handle the hero bridge mint
+    // executeMessage() handles permissioning checks
+    function _handleMessage(
```

Comment:
```suggestion
    function _handleMessage(bytes32, uint256, bytes memory _message, address)
```
If a variable isn't used, remove its name from function definition to silence the linter.